test/features/tbdex/tbdex_quote_notifier_test.dart

```diff
@@ -0,0 +1,59 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mocktail/mocktail.dart';
+import '../../helpers/mocks.dart';
+import '../../helpers/test_data.dart';
+
+void main() async {
+  late MockTbdexService mockTbdexService;
+  late MockTbdexQuoteNotifier mockTbdexQuoteNotifier;
+
+  await TestData.initializeDids();
+
+  setUpAll(() {
+    registerFallbackValue(MockBearerDid());
+    registerFallbackValue(TestData.aliceDid);
+  });
+
+  setUp(() {
+    mockTbdexService = MockTbdexService();
+    mockTbdexQuoteNotifier = MockTbdexQuoteNotifier();
+
+    when(() => mockTbdexService.getExchange(any(), any(), any()))
+        .thenAnswer((_) async => TestData.getExchange());
+  });
+
+  group('TbdexQuoteNotifier', () {
+    test('should handle high-frequency polling with exponential backoff',
+        () async {
+      when(() => mockTbdexQuoteNotifier.pollForQuote(any(), any()))
+          .thenAnswer((_) async => TestData.getQuote());
+
+      await mockTbdexQuoteNotifier.startPolling('pfiId', 'exchangeId');
+
+      verify(() => mockTbdexQuoteNotifier.startPolling('pfiId', 'exchangeId'))
+          .called(1);
+    });
```

Comment:
if you want to test the exponential backoff, try verifying how many times `getExchange()` is called